mm2src/mm2_main/src/lp_ordermatch/orderbook_rpc.rs

```diff
@@ -107,8 +108,11 @@ fn build_aggregated_entries_v2(
     (aggregated, total_base.into(), total_rel.into())
 }
 
-pub fn is_my_order(my_pub: &Option<String>, order_pubkey: &str) -> bool {
-    my_pub.as_ref().map(|my| my == order_pubkey).unwrap_or(false)
+// ZHTLC protocol coin uses random keypair to sign P2P messages per every order.
+// So, each ZHTLC order has unique «pubkey» field that doesn’t match node persistent pubkey derived from passphrase.
+// We can compare pubkeys from maker_orders and from asks or bids, to find our order.
+pub fn is_my_order(my_orders_pubkeys: &HashSet<String>, my_pub: &Option<String>, order_pubkey: &str) -> bool {
```

Comment:
It's only used in the current module, therefore we can avoid defining the function as `pub`. It could be also great if you can inline this function.

Reply:
oh, I see, this function should be private.
But in this case, its not so necessary to inline private function according to Kladov's explanation and recommendations 
As main usage of #[inline] comes from — it enables cross-crate inlining.
[it usually isn’t necessary to apply #[inline] to private functions — within a crate, the compiler generally makes good inline decisions.](https://matklad.github.io/2021/07/09/inline-in-rust.html). There’s [a joke](https://twitter.com/ManishEarth/status/936084757212946432) that LLVM’s heuristic for when the function should be inlined is “yes”.

Please, correct me, if I'm wrong in my understanding.

Reply:
> oh, I see, this function should be private. But in this case, its not so necessary to inline private function according to Kladov's explanation and recommendations As main usage of #[inline] comes from — it enables cross-crate inlining. [it usually isn’t necessary to apply #[inline] to private functions — within a crate, the compiler generally makes good inline decisions.](https://matklad.github.io/2021/07/09/inline-in-rust.html). There’s [a joke](https://twitter.com/ManishEarth/status/936084757212946432) that LLVM’s heuristic for when the function should be inlined is “yes”.
> 
> Please, correct me, if I'm wrong in my understanding.

If you want to make an order to compiler that function should be inlined, then you use `#[inline(always)]`.

If you want to suggest to compiler that function should be inlined, then you use `#[inline]` without any flag.

If you want to make an order to compiler that function should never be inlined, then you use `#[inline(never)]`

If the function is public, without inline/inline(always) attribute it will never be inlined to external crates unless you are using link time optimizations.
But that doesn't mean all in-crate functions are inlined. Inline attributes are useful even for private functions if needed.

Anyway, in this example because of the function context is all about comparison, I can tell you this function will be inlined by the compiler even if you don't give an attribute.
But I personally start giving inline suggestions(or using always flag for strict order) to compiler for even private functions. Because as I said it might not be inlined without giving attribute, and in case you will need to do some compiler or binary level optimizations, you will have less magical/high-level behaviours from compiler without having any knowladge.

Reply:
Oh, got your point. I added #[inline(always)].